modules/ximgproc/include/opencv2/ximgproc/segmentation.hpp

```diff
@@ -117,6 +296,67 @@ namespace cv {
 
             };
 
+            // SelectiveSearch helpers
```

Comment:
It is better to move implementation-specific declarations from public header file. You can place them in .cpp file.